### DISCUSSION TITLE
Run command externally and exit without prompt

### DISCUSSION THREAD

**Discussion by peternepp** [cmderdev/cmder#2541] [Q&A]
Hi there,

first of all thanks for the amazing Cmder. I'm quite comfortable with Cmder, using it for years now and it's becoming a daily driver for my work. 👍 

I wonder if there's a way to call an arbitrary command within a pre-defined ConEmu/Cmder task (cmd.exe) from another process without any prompt? I've tried so far:

``` 
cmder /x "-run {cmd::myTask} && python -m myModule && exit 0"
```

Altogether it's working, but the attempt of quitting cmd.exe with "exit 0" is prompting "Press Enter or Esc to exit" before I've to close the shell manually.

``` 
cmder /x "-run {cmd::myTask} && python -m myModule -exit"
```

Doesn't work. "-exit" will be recognized as attribute for the python command.

``` 
cmder /x "-run {cmd::myTask} && python -m myModule" /x "-exit"
```

``` 
cmder /x "-exit -run {cmd::myTask} && python -m myModule"
```

Both attempts are exiting immediately without running the task nor the command.

Any suggestions?

Many thanks,
Peter

**Comment by mrx23dot** [comment]
The easiest non cmder solution is to create a bin folder, add to path, add your scripts to mycommand.bat files.
Type mycommand

**Comment by GlassGruber** [comment]
A bit late, but I think you can use `-cur_console:n` where `n -- disable ‘Press Enter or Esc to close console’`.

I've tested with below and I think it works as you need.
```
Cmder.exe -- -run "{Cmder} -cur_console:n && exit 0"
```
You can find more in the [docs](https://conemu.github.io/en/NewConsole.html#syntax).

You can even test with more options like `-MinTSA --  Start ConEmu in TSA, but minimize to taskbar after restoring. Also ConEmu will not be exited after last console close, but hided to TSA.` from [here](https://conemu.github.io/en/ConEmuArgs.html#conemuexe-command-line-switches).

Thank you @peternepp because I was looking how to call cmder from an external program and your examples helped a lot!
